### PR TITLE
ROC-4014: more time requested Json RPA implementation

### PR DESCRIPTION
**Pull request by singh-surjeet**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-4014

### Change description ###

More time requested Json RPA implementation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    @Captor
+    private ArgumentCaptor<EmailData> emailDataArgument;
+
+    private Claim claim;
+
+    private MoreTimeRequestedEvent event;
+
+    @Before
+    public void setUp() {
+        claim = SampleClaim
```
Comment (geek-soft) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java]: Please make claim local variable. No reason to make it class field.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            .builder()
+            .withRespondedAt(LocalDateTime.of(2018, 4, 26, 1, 1))
+            .build();
+
+        event = new MoreTimeRequestedEvent(claim, LocalDateTime.now().toLocalDate(), "<any-email>");
+
```
Comment (geek-soft) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java]: Please remove blank line.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationService.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @EventListener
+    public void notifyRobotics(MoreTimeRequestedEvent event) {
+        requireNonNull(event);
+
+        EmailData emailData = prepareEmailData(event.getClaim());
+        emailService.sendEmail(emailProperties.getSender(), emailData);
+    }
+
+    private EmailData prepareEmailData(Claim claim) {
+
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationService.java]: Please remove blank line.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        service.notifyRobotics(event);
+
+        verify(emailService).sendEmail(senderArgument.capture(), emailDataArgument.capture());
+
+        EmailAttachment moreTimeReqJsonAttachment = emailDataArgument.getValue().getAttachments().get(0);
+
+        String expectedJson = buildJsonMoreTimeRequestedFileBaseName(claim.getReferenceNumber()) + JSON_EXTENSION;
+
+        assertThat(moreTimeReqJsonAttachment.getContentType()).isEqualTo(MediaType.APPLICATION_JSON_VALUE);
+        assertThat(moreTimeReqJsonAttachment.getFilename()).isEqualTo(expectedJson);
```
Comment (sabahirfan) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java]: these assertions can be added to the previous test, we won't require to write a separate test for same results of a method call

Comment (singh-surjeet) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java]: It's deliberate separation of not doing too much in one test. One is meant for all values without attachments and second one is with attachment only.

Comment (singh-surjeet) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java]: It was deliberate action of not putting too much into one test and kept it separate purposefully.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java (hunk 1 above)
Comment (singh-surjeet) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java]: claim is required for more-time-req event which is used in both tests.

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java (hunk 2 above)
Comment (singh-surjeet) [reply src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java]: done

**Pull request merged**